Gems/Atom/RHI/Code/Source/RHI/FreeListAllocator.cpp

```diff
@@ -345,6 +350,28 @@ namespace AZ
             }
         }
 
+        float FreeListAllocator::ComputeFragmentation() const
+        {
+            if (m_byteCountTotal == 0 || m_descriptor.m_capacityInBytes == static_cast<size_t>(-1))
+            {
+                return 0.f;
+            }
+
+            size_t maxFreeBlockSize = 0;
+            NodeHandle handle = GetFirstFreeHandle();
+            while (handle.IsValid())
+            {
+                const Node& node = GetNode(handle);
+                if (node.m_size > maxFreeBlockSize)
+                {
+                    maxFreeBlockSize = node.m_size;
+                }
+                handle = node.m_nextFree;
+            }
+
+            return 1.f - static_cast<float>(maxFreeBlockSize) / (m_descriptor.m_capacityInBytes - m_byteCountTotal);
```

Comment:
Maybe add a check for division by 0?

Reply:
Thanks for the review, I added an earlier check to just return 0 if the bytes allocated matches the capacity.